lib/origen/power_domains.rb

```diff
@@ -0,0 +1,29 @@
+require 'colorize'
+require_relative './power_domains/power_domain'
+require_relative './power_domains/power_domains_collection'
+module Origen
+  module PowerDomains
+    def power_domains(expr = nil)
+      if expr.nil?
+        if @_power_domains.nil?
```

Comment:
Don't think you should return the keys here, this whole top half of the if branch could just be written as:

~~~ruby
@_power_domains ||= PowerDomainsCollection.new
~~~